Mathlib/SetTheory/Game/PGame.lean

```diff
@@ -1904,6 +1904,87 @@ theorem lt_iff_sub_pos {x y : PGame} : x < y ↔ 0 < y - x :=
       ⟩
 #align pgame.lt_iff_sub_pos SetTheory.PGame.lt_iff_sub_pos
 
+/-! ### Inserting an option -/
+
+/-- The pregame constructed by inserting `x'` as a new left option into x. -/
+def insertLeft (x x' : PGame.{u}) : PGame :=
+  match x with
+  | mk xl xr xL xR => mk (xl ⊕ PUnit) xr (Sum.elim xL fun _ => x') xR
```

Comment:
Any reason to not instead use
```suggestion
  | mk xl xr xL xR => mk (Option xl) xr (Option.elim xL fun _ => x') xR
```
or whatever the spelling is that works?

Reply:
This nicely avoids having any useless `_ : Unit` variables floating around

Reply:
I'm not sure about this. First, the proof of `insertLeft_numeric` seems to get a little more complicated, since `simp` doesn't seem to work very well with `Option` values and I have to do an additional layer of `constructor` / `rcases`. But also, perhaps more significantly, this wouldn't generalize to the case where we want to add several options at once, such as a whole family of options `xl' -> PGame`, in which case we definitely need a `xl ⊕ xl'`. I think we would want to do that later in the development of the theory.